nvaccess/nvda
# Refocus multi-category settings dialog with the provided category.

### Link to issue number:

Fix-up of #12800.
Described in #12800 (comment), paragraph "Test almost successful", "Test 1".

### Summary of the issue:
If the multi-category settings dialog is already opened and you try to open it a second time, it will be refocused. But the new expected category panel is not shown; the old panel remains shown instead.

### Description of how this pull request fixes the issue:
Change the panel when refocusing a MultiCategorySettingsDialog according to the panel passed as parameter.

### Testing strategy:
#### Test 1:
Press successively NVDA+control+G, NVDA+control+O, NVDA+control+M, NVDA+control+K.
Checked that the expected panel is shown each time.

#### Test 2:
* Press NVDA+control+D
* Change a setting (e.g. report font size)
* Press NVDA+control+K
* Change a setting (e.g. skim reading)
* Press enter to validate the settings dialog
* Reopen each setting panel to check that the options have been saved correctly in the config.

#### Test 3:
Press NVDA+Control+K, Alt+Tab to Windows Explorer (or whatever), open NVDA menu -> Preferences -> Settings.
Settings dialog is refocused on keyboard settings panel.

#### Test 4:
1. Press NVDA+control+V
2. tab to "Change" button and press it
3. Alt+Tab to Windows Explorer (or whatever)
4. Press NVDA+control+G
5. Alt+Tab to settings dialog
6. Press Escape

Result:
* At step 4., nothing happens. This issue is not new to this PR (already present in #12800). I think it should be fixed but can be acceptable however, i.e. #12800 should not be reverted just for this.
* At step 6. the focus is reported on the "Change" button (from Speech settings). But in realty, the General settings panel is shown.
I think that this last point should be resolved before merging this PR.

All tests are successful except test 4.

### Known issues with pull request:
None

### Code Review Checklist:

  - description is up to date
  - change log entries
  - Unit tests
  - System (end to end) tests
  - Manual testing
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
